primitives/io/src/lib.rs

```diff
@@ -774,25 +774,25 @@ pub trait Logging {
 /// Interface that provides functions for benchmarking the runtime.
 #[runtime_interface]
 pub trait Benchmarking {
-		/// Get the number of nanoseconds passed since the UNIX epoch
-		///
-		/// WARNING! This is a non-deterministic call. Do not use this within
-		/// consensus critical logic.
-		fn current_time() -> u128 {
-			std::time::SystemTime::now().duration_since(std::time::SystemTime::UNIX_EPOCH)
-				.expect("Unix time doesn't go backwards; qed")
-				.as_nanos()
-		}
+	/// Get the number of nanoseconds passed since the UNIX epoch
+	///
+	/// WARNING! This is a non-deterministic call. Do not use this within
+	/// consensus critical logic.
+	fn current_time() -> u128 {
```

Comment:
This is pretty much the same as `offchain::now`

Reply:
Not sure. Probably @shawntabrizi knows the reasoning behind this.

Reply:
Mainly I needed nanoseconds and did not want to touch the offchain worker stuff. Is there some downside to exposing my own host function?

Especially given the fact that we will eventually hide all these host functions behind a feature flag?

Reply:
Fair enough, I wanted to avoid duplicating functions (especially if you decide to switch to capabilities anyway). Increased precision and limited usage scope is a good argument to keep it as is though.